internal/sbi/consumer/udr_service.go

```diff
@@ -43,6 +43,183 @@ func (s *nudrService) getDataSubscription(uri string) *Nudr_DataRepository.APICl
 	return client
 }
 
+func (s *nudrService) GetSessionManagementPolicyData(uri string,
+	ueId string, sliceInfo *models.Snssai, dnn string) (
+	resp *DataRepository.ReadSessionManagementPolicyDataResponse,
+	problemDetails *models.ProblemDetails, err error,
+) {
+	if uri == "" {
+		problemDetail := util.GetProblemDetail("Can't find any UDR which supported to this PCF",
+			"GetSessionManagementPolicyData Can't find UDR URI")
+		return nil, &problemDetail, nil
+	}
+
+	if ueId == "" {
+		problemDetail := util.GetProblemDetail("Can't find any UDR which supported to this PCF",
+			"GetSessionManagementPolicyData Can't find UE ID")
+		return nil, &problemDetail, nil
+	}
+
+	if sliceInfo == nil {
+		problemDetails := util.GetProblemDetail("Can't find any UDR which supported to this PCF",
+			"GetSessionManagementPolicyData Can't find Slice Info")
+		return nil, &problemDetails, nil
+	}
+
+	if dnn == "" {
+		problemDetails := util.GetProblemDetail("Can't find any UDR which supported to this PCF",
+			"GetSessionManagementPolicyData Can't find DNN")
+		return nil, &problemDetails, nil
+	}
+
+	ctx, pd, err := s.consumer.Context().GetTokenCtx(models.ServiceName_NUDR_DR, models.NrfNfManagementNfType_UDR)
+	if err != nil {
+		return nil, nil, err
+	} else if pd != nil {
+		return nil, pd, nil
+	}
+
+	client := s.getDataSubscription(uri)
+	param := DataRepository.ReadSessionManagementPolicyDataRequest{
+		UeId:   &ueId,
+		Snssai: sliceInfo,
+		Dnn:    &dnn,
+	}
+	resp, err = client.SessionManagementPolicyDataDocumentApi.ReadSessionManagementPolicyData(ctx, &param)
+	if err == nil {
+		return resp, nil, nil
+	}
+	problem := err.(openapi.GenericOpenAPIError).Model().(models.ProblemDetails)
```

Comment:
You can't do this. 
**Must use type assertion!** 

There are many place use this, please modify it!